exporter/elasticsearchexporter/elasticsearch_bulk.go

```diff
@@ -175,70 +164,74 @@ func newBulkIndexer(logger *zap.Logger, client *elasticsearch7.Client, config *C
 		// See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32344
 		maxDocRetry = config.Retry.MaxRequests - 1
 	}
-
-	pool := &bulkIndexerPool{
-		wg:    sync.WaitGroup{},
-		items: make(chan esBulkIndexerItem, config.NumWorkers),
-		stats: bulkIndexerStats{},
+	manager.pool = &sync.Pool{
+		New: func() any {
+			bi, err := docappender.NewBulkIndexer(docappender.BulkIndexerConfig{
+				Client:                client,
+				MaxDocumentRetries:    maxDocRetry,
+				Pipeline:              config.Pipeline,
+				RetryOnDocumentStatus: config.Retry.RetryOnStatus,
+			})
+			if err != nil {
+				return fmt.Errorf("error creating docappender bulk indexer: %w", err)
+			}
+			return bi
+		},
 	}
-	pool.wg.Add(numWorkers)
-
-	for i := 0; i < numWorkers; i++ {
-		bi, err := docappender.NewBulkIndexer(docappender.BulkIndexerConfig{
-			Client:                client,
-			MaxDocumentRetries:    maxDocRetry,
-			Pipeline:              config.Pipeline,
-			RetryOnDocumentStatus: config.Retry.RetryOnStatus,
-		})
-		if err != nil {
-			return nil, err
-		}
-		w := worker{
-			indexer:       bi,
-			items:         pool.items,
-			flushInterval: flushInterval,
-			flushTimeout:  config.Timeout,
-			flushBytes:    flushBytes,
-			logger:        logger,
-			stats:         &pool.stats,
-		}
-		go func() {
-			defer pool.wg.Done()
-			w.run()
-		}()
+
+	// Create a bulk indexer once to validate the config options
+	if err, ok := manager.pool.Get().(error); ok {
+		return nil, err
 	}
-	return pool, nil
+
+	return manager, nil
 }
 
 type bulkIndexerStats struct {
 	docsIndexed atomic.Int64
 }
 
-type bulkIndexerPool struct {
-	items chan esBulkIndexerItem
-	wg    sync.WaitGroup
-	stats bulkIndexerStats
+type bulkIndexerManager struct {
+	closeCh chan struct{}
+	stats   bulkIndexerStats
+	logger  *zap.Logger
+	config  *Config
+	wg      *sync.WaitGroup
+	sem     *semaphore.Weighted
+	pool    *sync.Pool
```

Comment:
Any reason for introducing a sync pool? I would prefer to do it in a separate PR if possible to allow us to measure the performance improvements objectively. Also, as a follow up after https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33374 I am planning to introduce a pool for `Document` but I am reluctant to have too many pools in the code. In addition, `BulkIndexer` encapsulates a lot of complexity (and a lot of buffers) so I think we should objectively measure the performance of pooling the indexers.